### DISCUSSION TITLE
How to show a loading state when a lazy route is loading? (code-bsaed router)

### DISCUSSION THREAD

**Discussion by dev-pengi** [TanStack/router#2889] [Q&A]
i tried the pendingComponent but it only appear at the initial loading, after that when using in a bad internet conditions you have to wait some seconds untile the route is loaded and thepage appear, and can't show a loading UI

**Comment by schiller-manuel** [comment]
can you please explain this in more detail? loading the lazy route should only occur once, so why would you expect to get another loading state?

**Reply by dev-pengi** [reply]
when navigating to another page that hasn't been loaded yet, it just stays there for like 7 seconds loading the components and assets with no loading vsuals (tested with browser throttling 3G)

**Reply by schiller-manuel** [reply]
can you please provide a minimal complete example and describe all steps necessary to reproduce ?
